### ISSUE TITLE
Suggestion: Add a visualization script for model predictions (PR ready)

### ISSUE THREAD

**Issue by Linwei-Chen** [FengheTan9/U-Bench#13]
Hello,
To help with the qualitative analysis of model performance, I've developed a script (visualize.py) that generates a side-by-side comparison of the original image, its ground truth mask, and the model's prediction.
This feature allows for a more intuitive understanding of the model's segmentation results. I have already implemented this and submitted a pull request for review.
You can find the implementation details and code in the corresponding Pull Request: [#PR](https://github.com/FengheTan9/U-Bench/pull/12)
I would appreciate it if you could take a look. Thank you

This pull request introduces a new script, visualize.py, to generate visual comparisons of the model's predictions against their corresponding ground truth labels.
Key Features
Side-by-Side Comparison: For each image in the validation set, it generates a high-quality plot showing the Original Image, the Ground Truth Mask, and the Model's Prediction.
Model Agnostic: Can be used with any trained model checkpoint by specifying the model architecture and path.
High-Quality Output: Saves each visualization as a PDF file, which is ideal for reports, presentations, and further analysis.
Easy to Use: Fully configurable via command-line arguments.
This script is an essential tool for qualitatively evaluating model performance, helping to intuitively understand where the model performs well and where it fails.
How to Use
To run the visualization, use the following command structure. Here is an example:
```
python visualize.py \
    --model UTANet_FADC \
    --model_path /path/to/your/checkpoint_best.pth \
    --base_dir /path/to/your/dataset/busi \
    --dataset_name busi \
    --save_path /path/to/your/visualization_results \
    --gpu 0
```
The script will iterate through the validation dataset and save the output PDF files into the directory specified by --save_path.

**Comment by Linwei-Chen** [comment]
<img width="1275" height="432" alt="Image" src="https://github.com/user-attachments/assets/684f63f6-32c7-49de-a494-48e8cc14a34e" />

**Comment by FengheTan9** [comment]
Merged🤗
